tools/parameter_tuning.py

```diff
@@ -0,0 +1,47 @@
+from sisyphus import Job, Task, tk
+from typing import Any, List, Tuple
+
+import numpy as np
+
+from i6_core.util import instanciate_delayed
+
+
+class PickOptimalParametersJob(Job):
+    """
+    Pick a set of optimal pickleable parameters based on their assigned (dynamic) score value.
+    """
+
+    def __init__(self, parameters: List[Tuple[Any]], values: List[tk.Variable], mode="minimize"):
+        """
+        :param parameters: list of tuples of parameters, must be pickleable
+        :param values: list of tk.Variables containing int or float, used to determine the best
+            set of parameters. Some calculations might be done using DelayedOps math.
```

Comment:
I don't exactly understand the meaning of `parameters` and `values` from this.

From the code, I see that `parameters` and `values` must have the same length. This should also be in the documentation.
Then I see that all tuples in `parameters` must have the same length. This should also be in the documentation.

From the code, I understand that you `values` is used to determine the the best index, and then `parameters[best_index]` is written to the output. From that logic, it seems a bit arbitrary that `parameters[i]` must be always a list and always of the same length, but this is a restriction due to how Sisyphus works. This should also be explained in the documentation.

Reply:
> but this is a restriction due to how Sisyphus works

The use case we have also does not makes sense otherwise, so that is okay.